Cant find_package(PugiXML REQUIRED) twice

This happens since 1.11.1 and is a regression because it makes some external packages fails the autopkgtests
e.g.
https://ci.debian.net/data/autopkgtest/testing/amd64/i/ismrmrd/9008335/log.gz
Because of code like this:
```
if(USE_SYSTEM_PUGIXML)
   find_package(PugiXML CONFIG)
   if (NOT PUGIXML_FOUND)
     find_dependency(PugiXML)
   endif()
endif()
```


cat CMakeLists.txt
```
project(test)
cmake_minimum_required(VERSION 3.18)

find_package(PugiXML REQUIRED)
find_package(PugiXML REQUIRED)
find_package(PugiXML REQUIRED)
find_package(PugiXML REQUIRED)
find_package(PugiXML REQUIRED)
```

```
cmake ..
-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/pugixml/pugixml-config.cmake:26 (add_library):
  add_library cannot create imported target "pugixml" because another target
  with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)


CMake Error at /usr/lib/x86_64-linux-gnu/cmake/pugixml/pugixml-config.cmake:26 (add_library):
  add_library cannot create imported target "pugixml" because another target
  with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:6 (find_package)


CMake Error at /usr/lib/x86_64-linux-gnu/cmake/pugixml/pugixml-config.cmake:26 (add_library):
  add_library cannot create imported target "pugixml" because another target
  with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:7 (find_package)


CMake Error at /usr/lib/x86_64-linux-gnu/cmake/pugixml/pugixml-config.cmake:26 (add_library):
  add_library cannot create imported target "pugixml" because another target
  with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:8 (find_package)


-- Configuring incomplete, errors occurred!
See also "/ismrmrd-1.4.2.1/foo/bar/foo/CMakeFiles/CMakeOutput.log".
```

--- a/scripts/pugixml-config.cmake.in
+++ b/scripts/pugixml-config.cmake.in
@@ -5,7 +5,7 @@ include("${CMAKE_CURRENT_LIST_DIR}/pugixml-targets.cmake")
 # If the user is not requiring 1.11 (either by explicitly requesting an older
 # version or not requesting one at all), provide the old imported target name
 # for compatibility.
-if (NOT DEFINED PACKAGE_FIND_VERSION OR PACKAGE_FIND_VERSION VERSION_LESS "1.11")
+if (NOT TARGET pugixml AND (NOT DEFINED PACKAGE_FIND_VERSION OR PACKAGE_FIND_VERSION VERSION_LESS "1.11"))
   add_library(pugixml INTERFACE IMPORTED)
   # Equivalent to target_link_libraries INTERFACE, but compatible with CMake 3.10
   set_target_properties(pugixml PROPERTIES INTERFACE_LINK_LIBRARIES pugixml::pugixml)

